### ISSUE TITLE
Create new Java project includes a default http trigger function

### ISSUE THREAD

**Issue by EricJizbaMSFT** [Microsoft/vscode-azurefunctions#81]
The file 'Function.java' should not exist and the user should just create a trigger themselves

**Comment by jdneo** [comment]
This can aligin with the js function project creation, but what I am concerning is that it could be strange if Java developers find there is no source file in the maven archetype.

**Comment by EricJizbaMSFT** [comment]
The background implementation of a feature should not affect the experience of that feature. VS Code users don't know or care that you're using the maven archetype in the background

**Comment by akaroml** [comment]
@EricJizbaMSFT Why is a project with no code useful to the users? Can you elaborate or provide some success cases?

**Comment by EricJizbaMSFT** [comment]
I agree a project with no code is not helpful. However, my problem with the existing experience is that I have no control over the default trigger that is created. In an ideal scenario the user creates a project and is immediately prompted for the first trigger, including:
1. function name
1. type (http, timer, blob, queue, etc.)
1. settings (like AuthorizationLevel for an http trigger)

The default `Function.java` is helpful for brand new users, but is likely 'throw-away' code for a user with any experience.

**Comment by akaroml** [comment]
Valid point, thanks. And two follow-ups:
1. Deleting code for experienced users shouldn't be much trouble. But finding a clue to what's next would be a blocker for beginners. We don't have many experienced users for Azure Functions at this stage.
2. Asking 3 more questions when creating a project in the command palette will make the process too long and too much.

I would recommend revisiting this issue when we decide to introduce graphical UI. Form is the right way to go when multiple entries need to be filled. For now, let's put this issue in backlog.

**Comment by EricJizbaMSFT** [comment]
For 1, it's not about how _difficult_ it is to delete code, it's about how _annoying_ it is to delete code.

The biggest reason I'm still debating this is because the Java experience is not consistent with other languages, which don't include code by default. If we're not going to be consistent, we should do it because the experience is without-a-doubt better (the current experience is _debatabely_ better).

We initially thought multiple prompts would confuse users (as you hypothesized), but we did customer interviews and found it was not a big problem. I don't see VS Code moving to graphical UI anytime soon and I don't think that's a reason to wait.

**Comment by EricJizbaMSFT** [comment]
Moving to next release

**Comment by brunoborges** [comment]
any progress on this?

**Comment by jdneo** [comment]
+ @andxu @Flanker32

**Comment by Flanker32** [comment]
@brunoborges We are now planning to add parameter to specify trigger type in function archetype, the design is still in discussing, when this feature done, we could ask users to select triggers they want in VSCode and generate correspond project with Java archetype. 
I'll update here once there are updates for archetype.

**Comment by brunoborges** [comment]
I'm not sure adding more parameters to the archetype will help. Perhaps the Azure Functions extension for VS Code, and then the toolkits for Eclipse and IntelliJ is what should enhance their support for Java, no?

Why not add code templates/macros to these plugins instead?

**Comment by Flanker32** [comment]
> I'm not sure adding more parameters to the archetype will help. Perhaps the Azure Functions extension for VS Code, and then the toolkits for Eclipse and IntelliJ is what should enhance their support for Java, no?
> 
> Why not add code templates/macros to these plugins instead?

Now VSCode using the archetype to initialize a function project, so we need to update it if we want to replace the default trigger when create a project. For the template, functions maven plugin do support create function from template with `azure-functions:add`, and VSCode extension has supported this feature within command `Azure Functions: Create Function...`

**Comment by ejizba** [comment]
For all languages _except_ Java, our extension will prompt the user to create a function at the same time they create a project (although they can "Skip for now" if they want). @Flanker32 if you just remove the HttpTrigger from the archetype or give us a flag to do so, that would be preferred

**Comment by brunoborges** [comment]
@ejizba in every Java IDE, there are wizards to create new projects and what not. They all come with some boilerplate code example that developers don't mind deleting. 

Is your feedback something based on recent survey/observations?

**Comment by ejizba** [comment]
I'm fine with boilerplate code - that's the whole point of templates. I'm just asking for some flexibility from the archetype so that we can design the wizard for our IDE. Here's the prompt for other languages:

<img width="429" alt="Screen Shot 2020-12-15 at 3 38 04 PM" src="https://user-images.githubusercontent.com/11282622/102285683-a025ca80-3eeb-11eb-96df-a601d32c1268.png">

Plenty of users will pick "Http Trigger" and still end up with boilerplate code just like already exists in the archetype (except they will have entered a function name instead of using the hard-coded "HttpExample"). Are you opposed to us prompting about trigger type and function name for some reason?

**Comment by brunoborges** [comment]
I'd suggest to have a separate clean archetype rather one with parameters.
